x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -869,3 +904,36 @@ function pipeMigrations(...migrations: AlertMigration[]): AlertMigration {
   return (doc: SavedObjectUnsanitizedDoc<RawRule>) =>
     migrations.reduce((migratedDoc, nextMigration) => nextMigration(migratedDoc), doc);
 }
+
+/**
+ * This creates a migration map that applies search source migrations to legacy es query rules
+ */
+function getEsQueryAlertSearchSourceMigrations(
+  searchSourceMigrations: MigrateFunctionsObject
+): MigrateFunctionsObject {
+  return mapValues<MigrateFunctionsObject, MigrateFunction>(
+    searchSourceMigrations,
+    (
+        migrateSerializedSearchSourceFields: MigrateFunction<SerializedSearchSourceFields>
+      ): MigrateFunction =>
+      (state) => {
+        const _state = state as { attributes: RawRule };
```

Comment:
**nit:** can we call this `doc` similar to other migrations so variable names are consistent with other migration?

```suggestion
      (doc: SavedObjectUnsanitizedDoc<RawRule>) => {
        const _state = state;
```

Reply:
Cannot cast `SerializableRecord` to `RawRule`.

Reply:
@Dmitriynj agree about the types, but I think naming the variable `doc` | `_doc` would increase consistency here. also doc is a better naming (we basically copied this from the first migration in other plugins, think it makes sense her to make it consistent)

Reply:
Renamed.